securedrop/source_app/main.py

```diff
@@ -221,8 +221,12 @@ def logout():
         if logged_in():
             msg = render_template('logout_flashed_message.html')
 
-            # clear the session after we render the message so it's localized
+            # Clear the session after we render the message so it's localized
+            # If a user specified a locale, save it and restore it
+            default_locale = getattr(config, 'DEFAULT_LOCALE', 'en_US')
```

Comment:
I think it would be enough to not set locale at all, even if DEFAULT_LOCALE does not exist, because [en_US will be used as a last resort](https://github.com/freedomofpress/securedrop/blob/develop/securedrop/i18n.py#L100). Unless you found a case where it fails ?

Reply:
True, it seems like using `g.locale` here would satisfy all concerns, it will always be defined and will have the default set via `g.locale = i18n.get_locale()`. Modifying...